crates/dojo-lang/src/compiler.rs

```diff
@@ -131,8 +143,37 @@ impl Compiler for DojoCompiler {
         };
 
         let mut compiled_classes: HashMap<String, (Felt, ContractClass)> = HashMap::new();
+        let list_selector =  get_allowed_libfuncs_list(&props, &unit);
 
         for (decl, class) in zip(contracts, classes) {
+            match class.validate_version_compatible(list_selector.clone()) {
+                Ok(()) => {},
+                Err(AllowedLibfuncsError::UnsupportedLibfunc {
+                    invalid_libfunc,
+                    allowed_libfuncs_list_name,
+                }) => {
+                    let contract_name = decl.submodule_id.name(db.upcast_mut());
+                    let diagnostic = formatdoc! {r#"
+                        Contract `{contract_name}` include `{invalid_libfunc}` function that is not allowed in the libfuncs list `{allowed_libfuncs_list_name}`.
+                        Please locate and remove this function from contract `{contract_name}` to ensure compatibility.
+                    "#};
+
+                    if props.allowed_libfuncs_deny {
+                        ws.config().ui().error(diagnostic);
+                    } else {
+                        ws.config().ui().warn(diagnostic);
+                    }
+                },
+                Err(e) => {
+                    return Err(e).with_context(|| {
+                        format!(
+                            "Failed to check allowed libfuncs for contract: {contract_name}",
+                            contract_name = decl.submodule_id.name(db.upcast_mut())
```

Comment:
```suggestion
                        format!(
                            "Failed to check allowed libfuncs for contract: {contract_name}"
```
re-use the `contract_name` already computed.